File: 0764-largest-plus-sign/0764-largest-plus-sign.js
/**
 * @param {number} n
 * @param {number[][]} mines
 * @return {number}
 */
var orderOfLargestPlusSign = function(n, mines) {
    let grid = Array.from({length:n}, () => Array(n).fill(n));
    for(let [x, y] of mines)
        grid[x][y] = 0;
    for (let i = 0; i < n; ++i) {
            let l = 0, r = 0, u = 0, d = 0;
            for (let j = 0, k = n - 1; j < n; ++j, --k) {
                grid[i][j] = Math.min(grid[i][j], l = (grid[i][j] === 0 ? 0 : l + 1));
                grid[i][k] = Math.min(grid[i][k], r = (grid[i][k] === 0 ? 0 : r + 1));
                grid[j][i] = Math.min(grid[j][i], u = (grid[j][i] === 0 ? 0 : u + 1));
                grid[k][i] = Math.min(grid[k][i], d = (grid[k][i] === 0 ? 0 : d + 1));
            }
        }
    let ans = 0;
        for (let i = 0; i < n; ++i) {
            for (let j = 0; j < n; ++j) {
                ans = Math.max(ans, grid[i][j]);
            }
        }

    return ans;
};

File: 0764-largest-plus-sign/0764-largest-plus-sign.py
class Solution:
    def orderOfLargestPlusSign(self, n: int, mines: List[List[int]]) -> int:
        dp = [[n]*n for _ in range(n)]
        for x, y in mines:
            dp[x][y] = 0
        
        for i in range(n):
            l = r = u = d = 0
            for j in range(n):
                k = n - j - 1
                l = 0 if dp[i][j] == 0 else l+1
                dp[i][j] = min(dp[i][j], l)
                r = 0 if dp[i][k] == 0 else r+1
                dp[i][k] = min(dp[i][k], r)
                u = 0 if dp[j][i] == 0 else u+1
                dp[j][i] = min(dp[j][i], u)
                d = 0 if dp[k][i] == 0 else d+1
                dp[k][i] = min(dp[k][i], d)
        
        ans = 0
        for i in range(n):
            for j in range(n):
                ans = max(ans, dp[i][j])
        
        return ans